src/material/tree/node.ts

```diff
@@ -25,36 +25,106 @@ import {
   booleanAttribute,
   numberAttribute,
 } from '@angular/core';
+import {CanDisable, HasTabIndex} from '@angular/material/core';
+import {BooleanInput, coerceBooleanProperty, coerceNumberProperty} from '@angular/cdk/coercion';
+import {LegacyTreeKeyManager, TreeKeyManagerItem, TreeKeyManagerStrategy} from '@angular/cdk/a11y';
+
+/**
+ * Determinte if argument TreeKeyManager is the LegacyTreeKeyManager. This function is safe to use with SSR.
+ */
+function isLegacyTreeKeyManager<T extends TreeKeyManagerItem>(
+  keyManager: TreeKeyManagerStrategy<T>,
+): keyManager is LegacyTreeKeyManager<T> {
+  return !!(keyManager as any)._isLegacyTreeKeyManager;
+}
 
 /**
  * Wrapper for the CdkTree node with Material design styles.
  */
 @Directive({
   selector: 'mat-tree-node',
   exportAs: 'matTreeNode',
+  outputs: ['activation', 'expandedChange'],
   providers: [{provide: CdkTreeNode, useExisting: MatTreeNode}],
   host: {
     'class': 'mat-tree-node',
+    '[attr.aria-expanded]': '_getAriaExpanded()',
```

Comment:
Same as with CDK tree, all of these function calls scare me. Lets make sure we have examples of using this in OnPush context and that they work properly